bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -66,11 +66,18 @@ struct CheckpointSummaryMsg : public BCStateTranBaseMsg {
   uint64_t requestMsgSeqNum;
 
   static bool equivalent(const CheckpointSummaryMsg* a, const CheckpointSummaryMsg* b) {
-    return (memcmp(a, b, sizeof(CheckpointSummaryMsg)) == 0);
+    return (a->lastBlock == b->lastBlock and
+      a->checkpointNum == b->checkpointNum and
+      (memcmp(&a->digestOfLastBlock, &b->digestOfLastBlock, sizeof(STDigest)) == 0) and
+      (memcmp(&a->digestOfResPagesDescriptor, &b->digestOfResPagesDescriptor, sizeof(STDigest)) == 0));
   }
 
   static bool equivalent(const CheckpointSummaryMsg* a, uint16_t a_id, const CheckpointSummaryMsg* b, uint16_t b_id) {
-    if (memcmp(a, b, sizeof(CheckpointSummaryMsg)) != 0) {
+    if (a->lastBlock != b->lastBlock ||
```

Comment:
If someone adds fields to this message in the future, and they should be compared too, we might have a bug.
I suggest adding a static assert here with this bool expression(pseudo-code):
`sizeof(CheckpointSummaryMsg) - sizeof(CheckpointSummaryMsg::requestMsgSeqNum == size of all compared elements`
Also, at the end add a warning. That way if class change the developer can decide if to compare the new field or  not.